manager/integration/tests/test_ha.py

```diff
@@ -2455,22 +2462,25 @@ def test_replica_failure_during_attaching(settings_reset, client, core_api, volu
     """
     Steps:
     1. Set a short interval for setting replica-replenishment-wait-interval.
-    2. Disable the setting soft-node-anti-affinity.
-    3. Create volume1 with 1 replica. and attach it to the host node.
-    4. Mount volume1 to a new mount point. then use it as an extra node disk.
-    5. Disable the scheduling for the default disk of the host node,
+    2. Enable the setting replica-node-soft-anti-affinity to allow scheduling
+       of a new replica to a node containing a failed replica.
+    3. Disable the setting replica-disk-soft-anti-affinity to disallow
+       scheduling of a new replica to a disk containing a failed replica.
+    4. Create volume1 with 1 replica. and attach it to the host node.
+    5. Mount volume1 to a new mount point. then use it as an extra node disk.
+    6. Disable the scheduling for the default disk of the host node,
        and make sure the extra disk is the only available disk on the node.
-    6. Create and attach volume2, then write data to volume2.
-    7. Detach volume2.
-    8. Directly unmount volume1 and remove the related mount point directory.
+    7. Create and attach volume2, then write data to volume2.
+    8. Detach volume2.
+    9. Directly unmount volume1 and remove the related mount point directory.
        --> Verify the extra disk becomes unavailable.
-    9. Attach volume2.
+    10. Attach volume2.
        --> Verify volume will be attached with state Degraded.
-    10. Wait for the replenishment interval.
+    11. Wait for the replenishment interval.
         --> Verify a new replica will be created but it cannot be scheduled.
-    11. Enable the default disk for the host node.
-    12. Wait for volume2 becoming Healthy.
-    13. Verify data content and r/w capability for volume2.
+    12. Enable the default disk for the host node.
```

Comment:
At this point, the purpose of the test is complete. We successfully recognized a failed replica during attaching and (after `replica-replenishment-wait-interval`) scheduled a new replica to replace it.

Previously, the test relied on `replica-node-soft-anti-affinity == false` to ensure the new replica could not schedule anywhere until the default disk on the test node was enabled. Once the default disk was enabled, the failed replica did not prevent scheduling to the node.

Now, the failed replica does prevent scheduling to the node. However, we can rely on `replica-disk-soft-anti-affinity == false` to continue to ensure it does not get scheduled until we enabled the default disk.